chartbeat.php

```diff
@@ -326,39 +351,60 @@ function loadChartbeat() {
 	}
 }
 
-// LOAD AMP analytics code
-function chartbeat_amp_add_analytics_head( $amp_template ) {
-?>
-  <script async custom-element="amp-analytics" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"></script>
-<?php
+
+function chartbeat_amp_add_analytics( $analytics ) {
+  if ( ! is_array( $analytics ) ) {
+      $analytics = array();
+  }
+	$user_id = get_option('chartbeat_userid');
+	if ($user_id) {
+		// if visitor is admin AND tracking is off, do not load chartbeat
+		if ( current_user_can( 'manage_options') && get_option('chartbeat_trackadmins') == 0)
+			return $analytics ;
+
+		$cb_configs = chartbeat_configs(); 
+
+		$analytics['chartbeat'] = array(
+	    'type' => 'chartbeat',
+	    'attributes' => array(),
+	    'config_data' => array(
+	        'vars' => array(
+	            'uid' => $cb_configs["uid"],
+	            'domain' => $cb_configs["domain"],
+	        )
+	    ),
+	  );
+
+	  $enable_newsbeat = get_option('chartbeat_enable_newsbeat');
+		if ($enable_newsbeat) { 
+			$analytics['chartbeat']['config_data']['vars']['authors'] = $cb_configs['author'];
+			$analytics['chartbeat']['config_data']['vars']['sections'] = $cb_configs['sections'];
+		}
+	}
+	return $analytics;
 }
 
-add_action( 'amp_post_template_head', 'chartbeat_amp_add_analytics_head' );
+add_filter( 'amp_post_template_analytics', 'chartbeat_amp_add_analytics' );
 
+function chartbeat_fbia_analytics( $analytics ) {
+	$user_id = get_option('chartbeat_userid');
+	if ($user_id) {
+		// if visitor is admin AND tracking is off, do not load chartbeat
+		if ( current_user_can( 'manage_options') && get_option('chartbeat_trackadmins') == 0)
+			return $analytics ;
 
-function chartbeat_amp_add_analytics_footer( $amp_template ) { 
-	$cb_configs = chartbeat_configs(); 
-?>
-	<amp-analytics type="chartbeat">
-	  <script type="application/json">
-	    {
-	      "vars": {
-	        "uid": <?php echo intval( $user_id ); ?>,
-<?php
-					$enable_newsbeat = get_option('chartbeat_enable_newsbeat');
-					if ($enable_newsbeat) { ?>
-	        "authors": "<?php echo $cb_configs["author"] ?>",
-	        "sections": "<?php echo $cb_configs["sections"] ?>",
-<?php } ?>
-	        "domain": "<?php echo $cb_configs["domain"]; ?>"
-	      }
-	    }
-	  </script>
-	</amp-analytics>
-<?php
+		add_chartbeat_config();
```

Comment:
The [FBIA spec requires](https://developers.facebook.com/docs/instant-articles/reference/analytics) that tracking code be wrapped which seems to be missing here, and would cause the feed to be invalid.